o.js

```diff
@@ -413,6 +413,14 @@
             return (base);
         }
 
+        // +++
+        // adds custom function to the path 
+        // +++
+        base.func = function (functionString) {
+            resource.path.push({ resource: functionString, get: null });
```

Comment:
Are you sure you don't need the `removeQuery('$format');` line? I'm not sure what that actually does.